packages/jellyfish-api-core/__tests__/category/account/removePoolLiquidity.test.ts

```diff
@@ -0,0 +1,108 @@
+    await container.start()
+    await container.waitForReady()
+    await container.waitForWalletCoinbaseMaturity()
+    await container.waitForWalletBalanceGTE(200)
+  })
+
+  afterAll(async () => {
+    await container.stop()
+  })
+
+  async function setup (symbolA: string, symbolB: string): Promise<string> {
+    const addressA = await getNewAddress(container)
+    const addressB = await getNewAddress(container)
+    const poolLiquidityAddress = await getNewAddress(container)
+
+    await createToken(container, symbolA)
+    await mintTokens(container, symbolA, { address: addressA })
+    await createToken(container, symbolB)
+    await mintTokens(container, symbolB, { address: addressB })
+    await createPoolPair(container, symbolA, symbolB)
+    await addPoolLiquidity(container, {
+      tokenA: symbolA,
+      amountA: 100,
+      tokenB: symbolB,
+      amountB: 100,
+      shareAddress: poolLiquidityAddress
+    })
+
+    return poolLiquidityAddress
+  }
+
+  it('should removeLiquidity', async () => {
+    const poolLiquidityAddress = await setup('DOG', 'FOX')
+
+    const poolPairBefore = (await container.call('getpoolpair', ['DOG-FOX']))['3']
+    const totalLiquidityBefore = new BigNumber(poolPairBefore.totalLiquidity)
+    const poolLiquidityAccountBefore = await container.call('getaccount', [poolLiquidityAddress]) // 44.72135954
+    expect(poolLiquidityAccountBefore.length).toStrictEqual(1)
+
+    const fee = new BigNumber('0.00001')
+    expect(poolLiquidityAccountBefore[0]).toStrictEqual(`${totalLiquidityBefore.minus(fee).toFixed(8)}@DOG-FOX`) // 44.72134954@DOG-FOX
+
+    const hash = await client.poolpair.removePoolLiquidity(poolLiquidityAddress, '13@DOG-FOX')
+    expect(typeof hash).toStrictEqual('string')
+    expect(hash.length).toStrictEqual(64)
+    await container.generate(1)
+
+    const poolPairAfter = (await container.call('getpoolpair', ['DOG-FOX']))['3']
+    const totalLiquidityAfter = new BigNumber(poolPairAfter.totalLiquidity)
+    expect(totalLiquidityBefore.minus(totalLiquidityAfter).toString()).toStrictEqual('13')
+
+    const poolLiquidityAccountAfter = await container.call('getaccount', [poolLiquidityAddress])
+    expect(poolLiquidityAccountAfter.length).toStrictEqual(3)
+
+    const dogLiquidityDiff = new BigNumber(poolPairBefore.reserveA).minus(poolPairAfter.reserveA)
+    const foxLiquidityDiff = new BigNumber(poolPairBefore.reserveB).minus(poolPairAfter.reserveB)
+    expect(poolLiquidityAccountAfter[0]).toStrictEqual(`${dogLiquidityDiff.toFixed(8)}@DOG`)
+    expect(poolLiquidityAccountAfter[1]).toStrictEqual(`${foxLiquidityDiff.toFixed(8)}@FOX`)
+    expect(poolLiquidityAccountAfter[2]).toStrictEqual(`${totalLiquidityAfter.minus(fee).toFixed(8)}@DOG-FOX`)
+  })
+
+  it('should removeLiquidity with utxos', async () => {
+    const poolLiquidityAddress = await setup('APE', 'BEE')
+    await container.fundAddress(poolLiquidityAddress, 25)
+    const utxos = await container.call('listunspent')
```

Comment:
Adopt the new approach here too. Don't use `listunspent`

Reply:
be noted here
here's telling the intention of adding utxo input
```
const rawtx = await container.call('getrawtransaction', [txid, true])

// the utxo input will be added into removeLiquidity tx vin
expect(rawtx.vin[0].txid).toStrictEqual(utxo.txid)
```

Reply:
I remove
```
    const utxos = await container.call('listunspent')
    const utxo = utxos.find((utxo: any) => utxo.address === poolLiquidityAddress)
```
The code will be like this
```
    const poolLiquidityAddress = await setup('APE', 'BEE')

    const { txid, vout } = await container.fundAddress(poolLiquidityAddress, 25)

    const txid1 = await client.poolpair.removePoolLiquidity(
      poolLiquidityAddress, '4@APE-BEE', { utxos: [{ txid, vout }] }
    )
    expect(typeof txid1).toStrictEqual('string')
    expect(txid1.length).toStrictEqual(64)

    await container.generate(1)

    const rawtx = await container.call('getrawtransaction', [txid1, true])

    // the utxo input will be added into removeLiquidity tx vin
    expect(rawtx.vin[0].txid).toStrictEqual(txid)
```